src/visual/Paper.js

```diff
@@ -43,12 +43,16 @@ export default class Paper extends joint.dia.Paper {
           return false;
         }
         // source
-        if (!magnetS || magnetS.getAttribute('class') !== 'port-body' ||
+        if (!magnetS || (magnetS.getAttribute('class') !== 'port-body' &&
+          magnetS.getAttribute('class') !== 'port-body-empty' &&
+          magnetS.getAttribute('class') !== 'port-body-empty available-magnet') ||
           magnetS.getAttribute('port-group') !== 'out') {
           return false;
         }
         // target
-        if (!magnetT || magnetT.getAttribute('class') !== 'port-body' ||
+        if (!magnetT || (magnetT.getAttribute('class') !== 'port-body' &&
+          magnetT.getAttribute('class') !== 'port-body-empty' &&
+          magnetT.getAttribute('class') !== 'port-body-empty available-magnet') ||
```

Comment:
Thank you so for this comment. I thought about it when wrote a code upper, but it was moddified old variant (I did not change this condition paradigm). Anyway thank you very much, this method has been refactored, according to your advice and was pushed with the d9b5b6b97c commit!